src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -165,12 +178,109 @@ def enable_autocomplete(self):
+            has_extension_name = False
+
+            if is_extension_command_source:
+                has_extension_name = hasattr(command_source, 'extension_name')
+
+            return is_extension_command_source and has_extension_name
+
+        # If the arguments have been processed into a namespace...
+        if isinstance(self._namespace, argparse.Namespace):
+            # Select the parsed command.
+            if hasattr(self._namespace, 'command'):
+                command = self._namespace.command
+        # Parse parameter names from user input.
+        if isinstance(self._raw_arguments, list):
+            raw_arguments = self._raw_arguments
+            parameters = extract_safe_params(self._raw_arguments)
+
+        for parameter in parameters:
+            parameter_set.add(parameter)
+
+        # If we can retrieve the extension from the current parser's command source...
+        if has_extension_name(self.command_source):
+            extension = self.command_source.extension_name
+        # Otherwise, the command may have not been in a command group. The command source will not be
+        # set in this case.
+        elif action and action.dest == '_subcommand':
+            # Get all parsers in the set of possible actions.
+            parsers = list(action.choices.values())
+            parser = parsers[0] if parsers else None
+            # If the first parser comes from an extension...
+            if parser and has_extension_name(parser.command_source):
+                # We're looking for a subcommand under an extension command group. Set the
+                # extension to reflect this.
+                extension = parser.command_source.extension_name
+            # Extend the command if the first raw argument is not a parameter.
+            if raw_arguments and raw_arguments[0] not in parameter_set:
+                command = '{cmd} {arg}'.format(cmd=command, arg=raw_arguments[0])
+        # Otherwise, only set the extension if every subparser comes from an extension. This occurs
+        # when an unrecognized argument is passed to a command from an extension.
+        elif isinstance(self.subparser_map, dict):
+            for _, subparser in self.subparser_map.items():
+                if isinstance(subparser.command_source, ExtensionCommandSource):
+                    extension = subparser.command_source.extension_name
+                else:
+                    extension = None
+                    break
+
+        if logger.isEnabledFor(logging.DEBUG):
+            import inspect
+            class_name = self.__class__.__name__
+            func_name = inspect.currentframe().f_code.co_name
+            format_str = '\n\t'.join([
+                f"{class_name}.{func_name}(command, parameters, extension):",
+                "Command: %s",
+                "Parameters: %s",
+                "Extension: %s"
+            ])
```

Comment:
Let's remove this part and makes it simple. We can add
```py
        logger.debug("recommendation_provider: version: %s, command: %s, parameters: %s, extension: %s",
                     version, command, parameters, extension)
```

to `recommendation_provider` instead.